assignments/lambda-classes.js

```diff
@@ -1 +1,149 @@
 // CODE here for your Lambda Classes
+
+// Person Class
+class Person {
+  constructor(attr) {
+    this.name = attr.name;
+    this.age = attr.age;
+    this.location = attr.location;
+    this.gender = attr.gender;
+  }
+  speak() {
+    return `Hello my name is ${this.name}, I am from ${this.location}.`;
+  }
+};
+
+// Instructor connects to Person
+class Instructor extends Person {
+  constructor(attr) {
+    super(attr);
+    // console.log(attr);
+    this.specialty = attr.specialty;
+    this.favLanguage = attr.favLanguage;
+    this.catchPhrase = attr.catchPhrase;
+  }
+  demo(subject) {
+    return `Today we are learning about ${subject}.`;
+  }
+  grade(student, subject) {
+    return `${student.name} receives a perfect score on ${subject}.`;
+  }
+}
+
+const siri = new Instructor({
+  name: 'Sirius Black ',
+  age: 45,
+  location: 'The Shadows',
+  gender: 'M',
+  specialty: 'Transfiguration',
+  favLanguage: 'JavaScript',
+  catchPhrase: `Keep on coding!!`
+});
+
+const luna = new Instructor({
+  name: 'Luna Lovegood',
+  age: 22,
+  location: 'Hogwarts',
+  gender: 'F',
+  specialty: 'Mystical Creatures',
+  favLanguage: 'JavaScript',
+  catchPhrase: `Look on the brightside of coding!!`
+});
+
+// Student connects to Person
+class Student extends Person {
+  constructor(attr) {
+    super(attr);
+    // console.log(attr);
+    this.previousBackground = attr.previousBackground;
+    this.className = attr.className;
+    this.favSubjects = attr.favSubjects;
+  }
+
+  listsSubjects() {
+    for (let i = 0; i < this.favSubjects.length; i++) {
+      console.log(`${this.name} favorite subject is ${this.favSubjects[i]}.`);
+    }
+  }
+
+  PRAssignment(subject) {
+    console.log(`${this.name} has submitted a PR for ${subject}.`);
+  }
+
+  sprintChallenge(subject) {
+    console.log(`${this.name} has begun sprint challenge on ${subject}.`);
+  }
+}
+
+const subject = 'JavaScript Fundamentals';
```

Comment:
const variable *subject* is not needed here. It can be passed as an argument to the methods above when invoking them.

Reply:
I left that in by accident, oopse. :D